test/integration_test/node_affinity_test.go

```diff
@@ -0,0 +1,82 @@
+// +build integrationtest
+
+package integrationtest
+
+import (
+	"testing"
+	"time"
+
+	op_corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultClusterName = "node-affinity-test"
+	defaultPxNamespace = "kube-system"
+
+	defaultValidateUninstallTimeout       = 15 * time.Minute
+	defaultValidateUninstallRetryInterval = 30 * time.Second
+)
+
+func TestNodeAffinity(t *testing.T) {
+	t.Run("testNodeAffinityLabels", testNodeAffinityLabels)
+}
+
+func testNodeAffinityLabels(t *testing.T) {
+	var err error
+	labelKey := "skip/px"
+	labelValue := "true"
+
+	// Construct Portworx StorageCluster object
+	cluster := &op_corev1.StorageCluster{}
+	cluster.Name = defaultClusterName
+	cluster.Namespace = defaultPxNamespace
+
+	// Set Node Affinity
+	cluster.Spec.Placement = &op_corev1.PlacementSpec{
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      labelKey,
+								Operator: v1.NodeSelectorOpNotIn,
+								Values:   []string{labelValue},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Get K8S nodes
+	nodeList, err := core.Instance().GetNodes()
+	require.NoError(t, err)
+
+	// Set affinity Label one of the K8S nodes
+	for _, node := range nodeList.Items {
```

Comment:
Make sure this is not a master node, else the test will fail. 
You can add this later when you add the validation for the test.

Reply:
Added check for master label.